lists.py

```diff
@@ -144,4 +145,41 @@ def balance_index(lst):
 
 
 assert balance_index([1, 2, 3, 6]) == 3
-assert balance_index([5, 2, 3, 1, 1, 1, 1]) == 2
\ No newline at end of file
+assert balance_index([5, 2, 3, 1, 1, 1, 1]) == 2
+
+
+"""
+In academia, the h-index is a metric used to calculate the impact of a researcher's papers. It is calculated as follows:
+A researcher has index h if at least h of her N papers have h citations each. If there are multiple h satisfying this 
+formula, the maximum is chosen.
+
+For example, suppose N = 5, and the respective citations of each paper are [4, 3, 0, 1, 5]. Then the h-index would be 3, 
+since the researcher has 3 papers with at least 3 citations.
+
+Given a list of paper citations of a researcher, calculate their h-index.
+"""
+
+
+def h_index(lst: List) -> int:
+    h, count = 0, 0
+    lst = sorted(lst)
+    for item in reversed(lst):
+        if count == h and count != 0:
+            return h
+        else:
+            h = item
+            count += 1
```

Comment:
Can we iterate on the list in a way that avoids this `count` variable and having to increment it manually?